integration/vite-css-build-test.ts

```diff
@@ -9,26 +12,44 @@ import {
 import type { Fixture, AppFixture } from "./helpers/create-fixture.js";
 import { PlaywrightFixture } from "./helpers/playwright-fixture.js";
 
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
+
 const TEST_PADDING_VALUE = "20px";
 
 test.describe("Vite CSS build", () => {
   let fixture: Fixture;
   let appFixture: AppFixture;
 
   test.beforeAll(async () => {
+    // set sideEffects for global vanilla extract css
+    let packageJson = JSON.parse(
+      await fs.readFile(
+        path.resolve(__dirname, "helpers", "node-template", "package.json"),
+        "utf-8"
+      )
+    );
+    packageJson.sideEffects = ["*.css.ts"];
```

Comment:
I realized that currently remix's template has `sideEffects: false` by default: 

https://github.com/remix-run/remix/blob/1d63d4ad46431058bd2de4b19ed70ddd17139cc0/templates/unstable-vite/package.json#L3

So, vanilla extract global style won't work out of the box.
However, I think this is a general behavior of `vite build` and not specific to remix, so probably any caveat should be mentioned on their documentation side, such as https://vanilla-extract.style/documentation/integrations/vite/